<!DOCTYPE html PUBLIC "-//W3C//DTD Html 1.0 Strict//EN" "http://www.w3.org/TR/Html1/DTD/strict.dtd"><html><head><title>Pigs</title></head><body><p><font face="Arial" color="#000100" size="2"></font><font face="Arial" color="#000100" size="3"><strong><img src="pigs0.bmp" alt="[pigs0]"></strong></font><font face="Arial" color="#0000FF" size="5"><strong>&nbsp;&nbsp;&nbsp;Pigs: genetic counseling and pedigree analysis<br></strong></font><font face="Arial" color="#000100" size="1"><br></font><font face="Arial" color="#000100" size="3"><br></font><font face="Arial" color="#000100" size="3">Spiegelhalter (1990) uses exact methods to analyse a small pedigree. This pedigree was previously used by Cannings and Thompson (1981) to illustrate their 'peeling' procedure to provide likelihoods for gene frequencies and probabilities for individuals being affected or carriers. We assume the pedigree refers to pigs which have the possibility of carrying a recessive gene: thus each pig has a genotype a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">, a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3"> or a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">, in which only those with a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3"> are affected with the trait, while those with a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3"> are carriers of the defective allele a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">. We assume that Ian (the consequence of a mating between Fred and his Aunt Clare) is yet to be born, and all that is known is that Fred's niece Jane has the trait. We wish to estimate the prevalence p of the allele a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">, and predict the chance of Ian being affected. The conditional probability distributions are as follows. For the genotype of the founder nodes Ann, Brian, Eric and Henry we assume a binomial distribution<br><br>&nbsp;&nbsp;&nbsp;Founder&nbsp;&nbsp;&nbsp;~&nbsp;&nbsp;&nbsp;Binomial(q, 2)<br><br>where Founder takes values 0, 1 or 2 for genotypes a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">, a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3"> and a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3"> respectively, and q = 1</font><font face="Symbol" color="#000100" size="3">- </font><font face="Arial" color="#000100" size="3">p is the prevalence of the allele a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">. This is equivalent to assuming Hardy Weinberg equilibrium, giving P(a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">) = q</font><font face="Arial" color="#000100" size="2"><sup>2</sup></font><font face="Arial" color="#000100" size="3">, P(a</font><font face="Arial" color="#000100" size="2"><sub>1</sub></font><font face="Arial" color="#000100" size="3">a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">) =2 q (1</font><font face="Symbol" color="#000100" size="3">- </font><font face="Arial" color="#000100" size="3">q) , P(a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3"> a</font><font face="Arial" color="#000100" size="2"><sub>2</sub></font><font face="Arial" color="#000100" size="3">) = (1</font><font face="Symbol" color="#000100" size="3">- </font><font face="Arial" color="#000100" size="3">q)</font><font face="Arial" color="#000100" size="2"><sup>2</sup></font><font face="Arial" color="#000100" size="3">.<br><br>For the genotype of offspring we have the standard Mendelian inheritance probabilities given by the following table. <br><center><br></font><font face="Arial" color="#000100" size="2"><img src="pigs1.bmp" alt="[pigs1]"><br></center></font><font face="Arial" color="#000100" size="3"><br>For a recessive gene the genotype-to-phenotype penetrance probabilities are given by:<br><center><br><br></font><font face="Arial" color="#000100" size="2"><img src="pigs2.bmp" alt="[pigs2]"><br></font><font face="Arial" color="#000100" size="3"></center>The necessary inheritance probabilities are read in from the data file as an array<br><br>BUGS code for Pigs model:</font><font face="Arial" color="#000100" size="3"><br></font><font face="Arial" color="#000100" size="1"><br></font><font face="Arial" color="#000100" size="3"></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">model <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">{<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">q ~ dunif(0,1) # prevalence of a1<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">p &lt;- 1 - q # prevalence of a2<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Ann1 ~ dbin(q,2); Ann &lt;- Ann1 + 1 # geno. dist. for founder<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Brian1 ~ dbin(q,2); Brian &lt;- Brian1 + 1 <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Clare ~ dcat(p.mendelian[Ann,Brian,]) # geno. dist. for child<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Diane ~ dcat(p.mendelian[Ann,Brian,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Eric1 ~ dbin(q,2)<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Eric &lt;- Eric1 + 1 <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Fred ~ dcat(p.mendelian[Diane,Eric,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Gene ~ dcat(p.mendelian[Diane,Eric,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Henry1 ~ dbin(q,2)<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Henry &lt;- Henry1 + 1 <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Ian ~ dcat(p.mendelian[Clare,Fred,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">Jane ~ dcat(p.mendelian[Gene,Henry,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">A1 ~ dcat(p.recessive[Ann,]) # phenotype distribution<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">B1 ~ dcat(p.recessive[Brian,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">C1 ~ dcat(p.recessive[Clare,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">D1 ~ dcat(p.recessive[Diane,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">E1 ~ dcat(p.recessive[Eric,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">F1 ~ dcat(p.recessive[Fred,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">G1 ~ dcat(p.recessive[Gene,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">H1 ~ dcat(p.recessive[Henry,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">I1 ~ dcat(p.recessive[Ian,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">J1 ~ dcat(p.recessive[Jane,]) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">a &lt;- equals(Ann, 2) # event that Ann is carrier<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">b &lt;- equals(Brian, 2) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">c &lt;- equals(Clare, 2) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">d &lt;- equals(Diane, 2) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">e &lt;- equals(Eric, 2) ;<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">f &lt;- equals(Fred, 2) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">g &lt;- equals(Gene, 2) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">h &lt;- equals(Henry, 2) <br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">for (J in 1:3) {<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">i[J] &lt;- equals(Ian, J) # i[1] = a1 a1<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3"># i[2] = a1 a2<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3"># i[3] = a2 a2 (i.e. Ian affected)<br></font><font face="Arial" color="#000100" size="1">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</font><font face="Arial" color="#000100" size="3">} </font><font face="Arial" color="#000100" size="1"><br></font><font face="Arial" color="#000100" size="3">&nbsp;&nbsp;&nbsp;}</font><font face="Arial" color="#000100" size="1"><br><br></font><font face="Arial" color="#000000" size="3"><br></font><font face="Arial" color="#0000FF" size="5"><a href="../Examples/Pigsdata.html"></font><font face="Arial" color="#0000FF" size="5"><u>Data</u></font><font face="Arial" color="#0000FF" size="5"></a></font><font face="Arial" color="#000000" size="3"> ( click to open )</font><font face="Arial" color="#000100" size="3"><br></font><font face="Arial" color="#0000FF" size="5"><br><a href="../Examples/Pigsinits.html"></font><font face="Arial" color="#0000FF" size="5"><u>Inits</u></font><font face="Arial" color="#0000FF" size="5"></a></font><font face="Arial" color="#000000" size="3"> ( click to open )</font><font face="Arial" color="#0000FF" size="5"><br></font><font face="Arial" color="#000100" size="1"><br></font><font face="Arial" color="#000100" size="1"><br></font><font face="Arial" color="#0000FF" size="5">Results <br></font><font face="Arial" color="#000000" size="3"><br>Time for 100000 updates 18s on 200MHz Pentium Pro. A 10000 update burn in followed by a further 100000 updates gave the parameter estimates<br><center><br></font><font face="Arial" color="#000000" size="2"><img src="pigs3.bmp" alt="[pigs3]"><br></font><font face="Arial" color="#000100" size="3"></center>We note a number of important tricks. First, each genotype is a 3-valued categorical variable with conditional probabilities either determined by the binomial (Hardy-Weinberg equilibrium) distribution (for founder nodes) or from the Mendelian inheritance probabilities which are stored as a 3-dimensional array </font><font face="Courier" color="#000100" size="3">p.mendelian</font><font face="Arial" color="#000100" size="3">. In the latter case, the genotype of the parents picks which row of the matrix is used for the distribution. However, the rows of this matrix are indexed by values 1, 2 or 3, whilst the genotypes of the founder nodes take values 0, 1 or 2. Since BUGS does not allow subscripts to be functions of variables, we must first add 1 to the genotype of the parents (for example, </font><font face="Courier" color="#000100" size="3">Ann = Ann1 + 1</font><font face="Arial" color="#000100" size="3">) and use these new variables as subscripts to the matrix </font><font face="Courier" color="#000100" size="3">p.mendelian</font><font face="Arial" color="#000100" size="3">. The genotype-to-phenotype distribution is handled similarly in a matrix </font><font face="Courier" color="#000100" size="3">p.recessive</font><font face="Arial" color="#000100" size="3">. Second, the </font><font face="Courier" color="#000100" size="3">equals</font><font face="Arial" color="#000100" size="3"> function </font><font face="Courier" color="#000100" size="3">equals(Ann, 2)</font><font face="Arial" color="#000100" size="3"> allows the calculation of P(Ann's genotype = 2) (i.e. a carrier), whilst </font><font face="Courier" color="#000100" size="3">equals(Ian, J)</font><font face="Arial" color="#000100" size="3"> calculates P(Ian's genotype = J), where J=3 implies that Ian is affected.</font></p></body></html>